src/kernel.cr

```diff
@@ -67,9 +67,227 @@ def printf(format_string, args : Array | Tuple) : Nil
   STDOUT.printf format_string, args
 end
 
-# Returns a formatted string.
+# Returns the string resulting from applying `format_string` to
+# any additional arguments.
+# Within the format string, any characters other than format sequences
+# are copied to the result.
 #
-# See also: `IO#printf`.
+# The syntax for a format sequence is:
+#
+# ```
+# %[flags][width][.precision]type
+# ```
+#
+# A format sequence consists of a percent sign, followed by optional flags,
+# width, and precision indicators, then terminated with a field type
+# character.  The field type controls how the corresponding
+# <code>sprintf</code> argument is to be interpreted, while the flags
+# modify that interpretation.
+#
+# The field type characters are:
+#
+#     Field |  Integer Format
```

Comment:
This will just produce clutter. I don't think there is a way to use a table in API docs. This needs to be improved (see #4613).

For now, these tables should be wrapped in plain code fences to use `<pre>` tag for formatting.